deeppavlov/pipeline_manager/plot_gen.py

```diff
@@ -0,0 +1,159 @@
+                                         "and it will not participate in comparing the results to display "
+                                         "the final plot.".format(key))
+            main[name][met] = met_max
+    return main
+
+
+def plot_res(info: dict,
+             name: str,
+             savepath: Union[str, Path],
+             save: bool = True,
+             width: float = 0.2,
+             fheight: int = 8,
+             fwidth: int = 12,
+             ext: str = 'png') -> None:
+    """
+    Creates a histogram with the results of various models based on the experiment log.
+
+    Args:
+        info: data for the plot
+        name: name of the plot
+        savepath: path to the folder where plot will be saved
+        save: determine save plot or show it without saving
+        width: width between columns of histogram
+        fheight: height of the plot
+        fwidth: width of the plot
+        ext: extension of the plot file
+
+    Returns:
+        None
+    """
+    from matplotlib import pyplot as plt
+
+    # prepeare data
+    bar_list = []
+    models = list(info.keys())
+    metrics = list(info[models[0]].keys())
+    n = len(metrics)
+
+    for met in metrics:
+        tmp = []
+        for model in models:
+            tmp.append(info[model][met])
+        bar_list.append(tmp)
+
+    x = np.arange(len(models))
+
+    # ploting
+    fig, ax = plt.subplots()
+    fig.set_figheight(fheight)
+    fig.set_figwidth(fwidth)
+
+    colors = plt.get_cmap('Paired')
+    colors = colors(np.linspace(0, 0.5, len(bar_list)))
+    # add some text for labels, title and axes ticks
+    ax.set_ylabel('Scores').set_fontsize(20)
+    ax.set_title('Scores by metric').set_fontsize(20)
+
+    bars = []
+    for i, y in enumerate(bar_list):
+        if i == 0:
+            bars.append(ax.bar(x, y, width, color=colors[i]))
+        else:
+            bars.append(ax.bar(x + i * width, y, width, color=colors[i]))
+
+    # plot x sticks and labels
+    ax.set_xticks(x - width / 2 + n * width / 2)
+    ax.set_xticklabels(tuple(models), fontsize=15)
+
+    yticks = ax.get_yticks()
+    ax.set_yticklabels(['{0:.2}'.format(float(y)) for y in yticks], fontsize=15)
+
+    ax.grid(True, linestyle='--', color='b', alpha=0.1)
+
+    # plot legend
+    # ax.legend(tuple([bar[0] for bar in bars]), tuple(metrics), loc='upper left', bbox_to_anchor=(1, 1))
+    ax.legend(tuple([bar[0] for bar in bars]), tuple(metrics))
+
+    # auto lables
+    def autolabel(columns):
+        for rects in columns:
+            for rect in rects:
+                height = rect.get_height()
+                ax.text(rect.get_x() + rect.get_width() / 2., 1.05 * height, '{0:.2}'.format(float(height)),
+                        ha='center', va='bottom', fontsize=12)
+
+    autolabel(bars)
+    plt.ylim(0, 1.1)
+
+    # show the picture
+    if not save:
+        plt.show()
+    else:
+        if not savepath.is_dir():
+            savepath.mkdir()
```

Comment:
Function signature shows that `savepath` could be an `str`, but you'll get an error in that case.